client/src/components/Attachment/UploadAttachment.js

```diff
@@ -105,10 +110,15 @@ const UploadAttachment = ({ getRelatedObject, edit, saveRelatedObject }) => {
   const handleFileEvent = async e => {
     // Control related object has an uuid or not
     if (!relatedObject.uuid) {
-      saveRelatedObject().then(async response => {
-        relatedObject.uuid = response
+      if (relatedObject.type !== "reports") {
+        relatedObject.uuid = saveRelatedObject()
```

Comment:
You should wait until the async `saveRelatedObject()` function has completed before saving the attachment.